### PR TITLE
Generate supportconfig for failed installations

### PR DESCRIPTION
**Pull request by ibrokethecloud**
This PR is related to issue: https://github.com/harvester/harvester/issues/1864

The aim is to auto generate a supportconfig when `debug` is enabled in the installation, and the installation fails.

The PR also introduces two custom plugins to copy the rke2.log and console.log into the generated supportconfig to make it easier to troubleshoot failures.

To test:
* patch /usr/sbin/harv-install to `exit 1`
* pass a minimal config via a http endpoint to enable debug. This would look as follows:
```yaml
install:
  debug: true
```
* on installation failure, harvester-installer will trigger the generation of support config, and also run the two packaged plugins

image

image

### CODE REVIEW

**Comment by bk201** [comment]
Probably we can move the installation failure messages to the end and suggest user that there is a support bundle.
Otherwise, the user might be a bit confused about why a support bundle is generated.

**Comment by ibrokethecloud** [comment]
@bk201 I have disabled logging of supportconfig to console.

Now we only provide info that supportconfig is being generated and location of same.

Sample output is as follows:

image

File: pkg/console/util.go [hunk 1]
```diff
@@ -448,6 +448,15 @@ func doInstall(g *gocui.Gui, hvstConfig *config.HarvesterConfig, webhooks Render
...
 		printToPanel(g, fmt.Sprintf(installFailureMessage, defaultLogFilePath), installPanel)
+		if hvstConfig.Debug {
+			printToPanel(g, "support config is being generated as running in debug mode, this can take a few minutes...", installPanel)
+			fileSuffix := fmt.Sprintf("harvester_%s", rand.String(5))
+			scErr := executeSupportconfig(ctx, fileSuffix)
+			if scErr != nil {
+				printToPanel(g, fmt.Sprintf("support bundle collection failed %v", err), installPanel)
+			}
+			printToPanel(g, fmt.Sprintf("support config is available at /var/log/scc_%s.txz", fileSuffix), installPanel)
+		}
```
Comment (bk201) [inline pkg/console/util.go:459]: LGTM.
nit: we can make the name more consistent. Either `Support config` or`Support bundle` for all strings.

Comment (ibrokethecloud) [reply pkg/console/util.go]: fixed it up

**Review by bk201 — APPROVED** [review]

**Review by johnliu55tw — APPROVED** [review]
LGTM

**Pull request merged**